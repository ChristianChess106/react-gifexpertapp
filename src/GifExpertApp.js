import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {


    const [categories, setCategories] = useState(['naruto']);


    return (
        <div>
                <h2>Hola Mundo</h2>

        <AddCategory setCategories={setCategories}/>

            <hr/>

        

            <ol>
                {
                    categories.map(category => {
                        return <GifGrid key={category} category={category}/>
                    })
                }
            </ol>
        </div>
    )
}
