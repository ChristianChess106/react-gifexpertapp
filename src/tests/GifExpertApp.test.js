import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Pruebas en gifexpertApp', () => {
    
    test('testeando gifExpertApp ', () => {
        
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de mostrar una lista de categorias ', () => {
        
        const categories = ["Naruto", "Dragon Ball"];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);

    });
    
})
