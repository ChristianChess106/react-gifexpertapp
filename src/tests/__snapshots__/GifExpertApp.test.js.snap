// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gifexpertApp debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    soy un componente funcionando
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en gifexpertApp testeando gifExpertApp  1`] = `
<Fragment>
  <h2>
    soy un componente funcionando
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
